src/fromager/commands/check_requirements_resolve.py

```diff
@@ -0,0 +1,59 @@
+import logging
+import sys
+
+import click
+from packaging.requirements import InvalidRequirement, Requirement
+
+from fromager import requirements_file, resolver, sources, wheels
+from fromager.context import WorkContext
+from fromager.requirements_file import RequirementType
+
+logger = logging.getLogger(__name__)
+
+
+@click.command()
+@click.argument(
+    "input_files_path", nargs=-1, required=True, type=click.Path(exists=False)
+)
+@click.pass_obj
+def check_requirements_resolve(wkctx: WorkContext, input_files_path: list[str]) -> None:
+    """
+    Check that top-level requirements resolve.
+    This command takes a single wildcard path string for constraints.txt and requirements.txt.
+    It checks the top-level entries of these files resolve and reports issues if found.
+    """
+    if not input_files_path:
+        logger.error("no constraints.txt or requirements.txt found in given paths")
+        sys.exit(1)
+
+    failure = False
+    for path in input_files_path:
+        for line in requirements_file.parse_requirements_file(path):
+            try:
+                req = Requirement(line)
+            except InvalidRequirement as err:
+                logger.error(f"{path}: invalid requirement {line!r}: {err}")
+                failure = True
+                continue
```

Comment:
The bootstrap command has some [code to get the requirements](https://github.com/python-wheel-build/fromager/blob/main/src/fromager/commands/bootstrap.py#L108), including filtering out requirements that are not relevant based on the current environment markers. How likely is it that we would need to do similar filtering in the linter?

Reply:
Our main aim with the lint command is to do an early and inexpensive check of what will be repeated behavior in the bootstrap command.  Given that, so long as it's inexpensive, it's probably best to keep the behavior identical in this regard.  I'll adjust it accordingly.